app/models/condition.rb

```diff
@@ -86,10 +86,7 @@ def self.evaluate(cond, rec, inputs={}, attr=:expression)
 
   def self.do_eval(expr)
     if expr =~ /^__start_ruby__\s?__start_context__\s?(.*)\s?__type__\s?(.*)\s?__end_context__\s?__start_script__\s?(.*)\s?__end_script__\s?__end_ruby__$/im
-      context, col_type, script = [$1, $2, $3]
-      context = MiqExpression.quote(context, col_type)
-      result = SafeNamespace.eval_script(script, context)
-      raise "Expected return value of true or false from ruby script but instead got result: [#{result.inspect}]" unless result.kind_of?(TrueClass) || result.kind_of?(FalseClass)
+      raise "Ruby script in expression is no longer supported"
```

Comment:
> Would deprecating mean leaving current implementation intact or just a fancier form of letting the user know that the functionality is gone?

Possibly.  Technically, we should warn a deprecation first now for the next release, and then completely remove it in the next-next release.  But, from a security/safety standpoint throwing it away and raise a warning as is done here seems ok for the next version, and then we completely throw it away in the next-next version (that is, even remove the code that says "__start_ruby__" and the raise itself.

In either case the alternative should be made clear to the user.  If "RegEx feature in conditions should cover off all advanced use cases for filtering." *is* the alternative, it should say that right in the error message.  Otherwise, how will the user know?